onnxruntime/core/providers/webnn/builders/impl/slice_op_builder.cc

```diff
@@ -127,22 +123,28 @@ bool SliceOpBuilder::IsOpSupportedImpl(const InitializedTensorSet& initializers,
     const auto data_type = steps_tensor.data_type();
     // WebNN doesn't support steps other than 1.
     if (data_type == ONNX_NAMESPACE::TensorProto_DataType_INT64) {
-      auto tensor_data = std::vector<int64_t>(reinterpret_cast<int64_t*>(unpacked_tensor.data()),
-                                              reinterpret_cast<int64_t*>(unpacked_tensor.data() +
-                                                                         unpacked_tensor.size()));
-      if (!std::all_of(tensor_data.begin(), tensor_data.end(), [](int64_t i) { return i == 1; })) {
+      if (!std::all_of(reinterpret_cast<int64_t*>(unpacked_tensor.data()),
+                       reinterpret_cast<int64_t*>(unpacked_tensor.data()) +
+                           unpacked_tensor.size() / sizeof(int64_t),
```

Comment:
```suggestion
                       reinterpret_cast<int64_t*>(unpacked_tensor.data() + unpacked_tensor.size()),
```

(minor) Could avoid the division by just adding the byte size first before the cast.